pennylane/utils.py

```diff
@@ -105,6 +106,56 @@ def decompose_hamiltonian(H, hide_identity=False):
     return coeffs, obs
 
 
+def sparse_hamiltonian(H):
+    r"""Computes the sparse matrix representation a Hamiltonian in the computational basis.
+
+    Args:
+        H (~.Hamiltonian): Hamiltonian operator for which the matrix representation should be
+         measured
+
+    Returns:
+        coo_matrix: a sparse matrix in scipy COOrdinate format with the dimension of
+        :math:`(2^n, 2^n)` where :math:`n` is the number of wires
+
+    **Example:**
+
+    This function can be used by passing a `qml.Hamiltonian` object as:
+
+    >>> coeffs = [0.5, 0.5]
```

Comment:
This is not possible yet, but in the future we should allow `coeffs` to be:

- a numpy array
- a tensorflow tensor
- a pytorch tensor

allowing for the Hamiltonian coefficients to be differentiable :)

Reply:
Thanks Josh!